packages/interface/src/index.ts

```diff
@@ -0,0 +1,61 @@
+type Mnemonic = string
+type Address = string
+enum Chain {
+  BITCOIN,
+  ETHEREUM,
+  NEAR,
+  TERRA,
+  SOLANA
+}
+
+enum Network {
+  MAINNET,
+  TESTNET
+}
+
+enum Token {
+  NATIVE,
+  ERC20,
+  ERC721
+}
+
+enum Hardware {
+  LEDGER
+}
+
+export default interface IWalletConstructor {
+  new (mnemomic: Mnemonic, network: Network): IWallet
+}
+
+export interface IWallet {
+  _mnemonic: Mnemonic
+  _network: Network
+  addAccount(chain: Chain, hardware?: Hardware): Account
+  getAccounts(chain: Chain): Account[]
+}
+
+type Account = {
+  _chain: Chain
+  _hardware?: Hardware
+  name: string
+  getAssets(): Promise<Asset[]>
+  getUnusedAddress(): Promise<Address>
+  getPublicKey(): Promise<string>
+  getPrivateKey(): Promise<string>
+}
+
+type Asset = {
+  type: Token
+  symbol: string
+  balance: BigNumber
+  getPastTransactions(): Promise<Transaction[]>
```

Comment:
Please note that we only store transactions locally as far as I know, and there is not a way now to retrieve them from the blockchain through the CAL? We can double check with the team about that.